modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/asset/model/DDMFormAssetRenderer.java

```diff
@@ -152,6 +157,38 @@ public String getTitle(Locale locale) {
 			false);
 	}
 
+	@Override
+	public PortletURL getURLEdit(
+			LiferayPortletRequest liferayPortletRequest,
+			LiferayPortletResponse liferayPortletResponse)
+		throws Exception {
+
+		HttpServletRequest httpServletRequest =
+			liferayPortletRequest.getHttpServletRequest();
+
+		String portletNamespace =
+			DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM_ADMIN;
+
+		PortletURL portletURL = _portal.getControlPanelPortletURL(
+			httpServletRequest, portletNamespace, PortletRequest.RENDER_PHASE);
+
+		portletURL.setParameter(
+			"mvcPath", "/admin/edit_form_instance_record.jsp");
+
+		portletURL.setParameter(
+			"formInstanceRecordId",
+			String.valueOf(_ddmFormInstanceRecord.getFormInstanceRecordId()));
+
+		portletURL.setParameter(
+			"formInstanceId",
+			String.valueOf(_ddmFormInstanceRecord.getFormInstanceId()));
+
+		portletURL.setParameter(
+			"redirect", _portal.getCurrentURL(httpServletRequest));
```

Comment:
@marcosapmf, the `redirect` parameter should be added following the same order as [here](https://github.com/natocesarrego/liferay-portal/blob/00c1907923252bee7f4cdef94cef7fa4b33f0963/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_record_action.jsp#L50-L53).